src/platform/backends/lxd/lxd_virtual_machine_factory.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "lxd_virtual_machine_factory.h"
+#include "lxd_request.h"
+#include "lxd_virtual_machine.h"
+
+#include <multipass/logging/log.h>
+
+#include <QFile>
+#include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QSslKey>
+
+#include <multipass/format.h>
+#include <yaml-cpp/yaml.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
+namespace
+{
+constexpr auto category = "lxd factory";
+} // namespace
+
+mp::LXDVirtualMachineFactory::LXDVirtualMachineFactory(const mp::Path& data_dir)
+    : data_dir{data_dir}, base_url{"https://10.225.118.1:8443/1.0"}, manager{std::make_unique<QNetworkAccessManager>()}
```

Comment:
I know the goal is to support https over Unix sockets, but for this PoC, using `localhost` here would make it easier to try out :smiley: